### ISSUE TITLE
Fix false positives for nested properties

### ISSUE THREAD

**Issue by Grawl** [stylelint/stylelint#3230]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug when Sass indented properties cause all selectors below as warnings.

> Which rule, if any, is this issue related to?

```yml
  no-duplicate-selectors: true
  indentation: tab
```

> What CSS is needed to reproduce this issue?

```sass
.foo
	a
		color: white
	background:
		color: black // this one
	%placeholder
		background: hsla(0, 0%, 79%, 0.2)
	&-bar
		color: white
.baz
	@extend %placeholder
	.qux
		color: red
```

> What stylelint configuration is needed to reproduce this issue?

```yml
rules:
  no-duplicate-selectors: true
  indentation: tab
```

> Which version of stylelint are you using?

```
➜ stylelint -v
9.1.3
```

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint path/to/stylesheet.sass`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Sass `.scss` indented syntax

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

Looks like every selector after indented property is marked as warning.

```
  6:1  ✖  Unexpected duplicate selector ".foo %placeholder ", first used at line 6   no-duplicate-selectors
  7:3  ✖  Expected indentation of 3 tabs                                             indentation
  8:1  ✖  Unexpected duplicate selector ".foo-bar ", first used at line 8            no-duplicate-selectors
  9:3  ✖  Expected indentation of 3 tabs                                             indentation
 10:1  ✖  Unexpected duplicate selector ".baz ", first used at line 10               no-duplicate-selectors
 12:1  ✖  Unexpected duplicate selector ".baz  .qux ", first used at line 12         no-duplicate-selectors
 12:2  ✖  Expected indentation of 2 tabs                                             indentation
 13:3  ✖  Expected indentation of 4 tabs                                             indentation
```

If I replace indented property to a regular one, the problem disappears

```sass
.foo
	a
		color: white
	background-color: black // this one
	%placeholder
		background: hsla(0, 0%, 79%, 0.2)
	&-bar
		color: white
.baz
	@extend %placeholder
	.qux
		color: red
```

**Comment by jeddy3** [comment]
@Grawl Thanks for the detailed report and for using the template. 

I'll label this up as a bug and help wanted. Unfortunately, I don't believe any core contributors use SASS syntax and there is only one maintainer of the `postcss-sass` parser. So, SASS support in stylelint relies on contributions both here and upstream. Please consider helping out if you can and if you'd like to see SASS support improve in stylelint.

Incidentally, [nested properties](http://sass-lang.com/documentation/file.SASS_REFERENCE.html#nested_properties) have proven to be a particularly difficult aspect of non-standard syntax to deal with. It might turn out, after an investigation, that `postcss-sass` is behaving as expected here and it is, in fact, stylelint's built-in rules that are tripping up. If this is the case, I believe there are three options available as the built-in rules as designed for standard syntax only:

1. Don't use nested properties.
2. Use nested properties but turn off all incompatible rules.
3. Write plugins of incompatible rules that understand nested properties.

**Comment by Grawl** [comment]
I really want to switch from sass-lint to stylelint but I am not a "real" programmer to send PRs, I just make a lot of websites using a lot of tools. I will send more issues like this in future, it's how I can help.

**Comment by vankop** [comment]
looks like it is fixed and we can close this
